core/src/main/java/org/apache/accumulo/core/iterators/OrIterator.java

```diff
@@ -80,59 +113,103 @@ public int compareTo(TermSource o) {
       // sorted after they have been determined to be valid.
       return this.iter.getTopKey().compareColumnQualifier(o.iter.getTopKey().getColumnQualifier());
```

Comment:
See the class-level javadoc "This Iterator will return a sorted iteration of docIDs for a given set of terms."

It's the same issue that I pinged @phrocker about. This is actually violating the SKVI contract (returning records out of order), but the implementation just happens to allow it.